File: apps/api/internal/handler/health.go
package handler

import "github.com/gofiber/fiber/v2"

// Health dipakai Cloud Run / Render untuk startup probe. Sengaja tidak
// menyentuh database supaya container tetap dianggap sehat saat Supabase
// free tier sedang di-resume dari auto-pause.
func (h *Handler) Health(c *fiber.Ctx) error {
	return c.JSON(fiber.Map{
		"status": "ok",
		"env":    h.cfg.Env,
	})
}

File: apps/api/src/routes/health.ts
import { Hono } from "hono";
import type { Env } from "../types";

export const health = new Hono<{ Bindings: Env }>();

// Dipakai sebagai startup/liveness probe. Sengaja tidak menyentuh database
// supaya Worker tetap dianggap sehat saat Supabase free tier sedang
// di-resume dari auto-pause.
health.get("/healthz", (c) => c.json({ status: "ok", env: c.env.APP_ENV }));
